crates/interledger-packet/src/address.rs

```diff
@@ -0,0 +1,385 @@
+//! ILP address types.
+//!
+//! Reference: [ILP Addresses - v2.0.0](https://github.com/interledger/rfcs/blob/master/0015-ilp-addresses/0015-ilp-addresses.md).
+
+// Addresses are never empty.
+#![allow(clippy::len_without_is_empty)]
+
+use std::error;
+use std::fmt;
+use std::str;
+
+use crate::errors::ParseError;
+use bytes::{BufMut, Bytes, BytesMut};
+use std::convert::TryFrom;
+use std::str::FromStr;
+
+const MAX_ADDRESS_LENGTH: usize = 1023;
+
+#[derive(Debug)]
+pub enum AddressError {
+    InvalidLength(usize),
+    InvalidFormat,
+}
+
+use std::error::Error;
+impl Error for AddressError {
+    fn description(&self) -> &str {
+        match *self {
+            AddressError::InvalidLength(length) => "invalid address length",
+            AddressError::InvalidFormat => "invalid address format",
```

Comment:
```suggestion
            AddressError::InvalidFormat => "Invalid address format",
```

Do you think it would be worth separating `InvalidCharacter` and `InvalidScheme` errors?

Reply:
If the address is malformed, the InvalidFormat error is sufficient for debugging I think. 
I doubt that more granularity is needed